src/Mod/Spreadsheet/App/PropertySheet.cpp

```diff
@@ -124,22 +127,34 @@ bool PropertySheet::isValidAlias(const std::string &candidate)
     if (ExpressionParser::isTokenAUnit(candidate))
         return false;
 
-    /* Check to make sure it doesn't match a cell reference */
-    if (boost::regex_match(candidate.c_str(), cm, gen)) {
-        static const boost::regex e("\\${0,1}([A-Z]{1,2})\\${0,1}([0-9]{1,5})");
-
-        if (boost::regex_match(candidate.c_str(), cm, e)) {
-            const boost::sub_match<const char *> colstr = cm[1];
-            const boost::sub_match<const char *> rowstr = cm[2];
+    App::Expression* ExpressionResult = 0;
 
-            // A valid cell address?
-            if (App::validRow(rowstr.str()) >= 0 && App::validColumn(colstr.str()) >= 0)
-                return false;
-        }
-        return true;
+    // check if the alias is a valid expression by parsing it
+    try {
+        ExpressionResult = App::ExpressionParser::parse(owner, candidate.c_str());
     }
-    else
+    catch (Base::Exception& exception) {
+        ExpressionResult = new App::StringExpression(owner, candidate.c_str());
+        //new App::DocumentObjectExecReturn(exception.what());
+        std::ostringstream ss;
+        ss << "Failed to adjust link for " << owner->getFullName() << " in expression "
+            << ExpressionResult->toString() << ": " << exception.what();
+        throw Base::RuntimeError(ss.str());
         return false;
```

Comment:
No need to return after throw, everything after return in the scope is dead code
```suggestion
```